frontend/test/metabase/scenarios/filters/filter-bulk.cy.spec.js

```diff
@@ -468,89 +431,67 @@ describe("scenarios > filters > bulk filtering", () => {
   describe("key filters", () => {
     beforeEach(() => {
       visitQuestionAdhoc(rawQuestionDetails);
-      openFilterModal();
+      filter();
     });
 
     it("filters by primary keys", () => {
-      modal().within(() => {
-        cy.findByLabelText("ID").within(() => {
-          cy.findByPlaceholderText("Enter an ID").type("17, 18");
-        });
-        cy.button("Apply").click();
+      filterField("ID", {
+        value: "17, 18",
       });
 
+      applyFilters();
+
       cy.findByText("Showing 2 rows").should("be.visible");
       cy.findByText("131.68").should("be.visible"); // total for order id 17
       cy.findByText("123.99").should("be.visible"); // total for order id 18
     });
 
     it("filters by a foreign key", () => {
-      modal().within(() => {
-        cy.findByLabelText("Product ID").within(() => {
-          cy.findByPlaceholderText("Enter an ID").type("65");
-        });
-        cy.button("Apply").click();
+      filterField("Product ID", {
+        value: "65",
       });
 
+      applyFilters();
+
       cy.findByText("Showing 107 rows").should("be.visible");
     });
   });
 
   describe("text filters", () => {
     beforeEach(() => {
       visitQuestionAdhoc(peopleQuestion);
-      openFilterModal();
+      filter();
     });
 
     it("adds a contains text filter", () => {
-      modal().within(() => {
-        cy.findByLabelText("City").within(() => {
-          cy.findByPlaceholderText("Search by City").type("Indian");
-        });
-        cy.button("Apply").click();
+      filterField("City", {
+        value: "Indian",
       });
+
+      applyFilters();
+
       cy.findByText("Showing 5 rows").should("be.visible");
     });
 
     it("adds an ends with text filter", () => {
-      modal().within(() => {
-        cy.findByLabelText("City").within(() => {
-          cy.findByText("Contains").click();
-        });
+      filterField("City", {
+        operator: "ends with",
+        value: "Valley",
       });
 
-      popover().within(() => {
-        cy.findByText("Ends with").click();
-      });
+      applyFilters();
 
-      modal().within(() => {
-        cy.findByLabelText("City").within(() => {
-          cy.findByPlaceholderText("Search by City").type("Valley");
-        });
-        cy.button("Apply").click();
-      });
       cy.findByText("Showing 8 rows").should("be.visible");
     });
 
     it("adds multiple is text filters", () => {
-      modal().within(() => {
-        cy.findByLabelText("City").within(() => {
-          cy.findByText("Contains").click();
-        });
+      filterField("City", {
+        operator: "is",
+        value: "Indianeown, Indian Valley",
       });
```

Comment:
by abstracting all this out to a helper function, we can potentially just change the helper if the UI changes, instead of touching all 50 tests that use the filter UI.